src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -496,20 +515,14 @@ function VisualizeAppController(
 
   $scope.onClearSavedQuery = () => {
     delete $scope.savedQuery;
-    delete $state.savedQuery;
-    $state.query = {
-      query: '',
-      language:
-        localStorage.get('kibana.userQueryLanguage') || uiSettings.get('search:queryLanguage'),
-    };
+    stateContainer.transitions.removeSavedQuery();
+    stateContainer.transitions.set('query', defaultQuery);
```

Comment:
I'm not 100% sure about the inner workings, but will this trigger two calls to connected subscribers of the state?

Reply:
Yes, this will trigger two calls. Better to batch in one transition if possible. 

p.s. : this still will cause only one browser history stack record.

Reply:
thanks, updated!